common/meta/types/src/user_stage.rs

```diff
@@ -178,12 +178,12 @@ impl UserStageInfo {
 }
 
 impl TryFrom<Vec<u8>> for UserStageInfo {
-    type Error = ErrorCode;
+    type Error = MetaError;
 
-    fn try_from(value: Vec<u8>) -> Result<Self> {
+    fn try_from(value: Vec<u8>) -> MetaResult<Self> {
         match serde_json::from_slice(&value) {
             Ok(info) => Ok(info),
-            Err(serialize_error) => Err(ErrorCode::IllegalUserInfoFormat(format!(
+            Err(serialize_error) => Err(MetaError::IllegalUserInfoFormat(format!(
                 "Cannot deserialize stage from bytes. cause {}",
                 serialize_error
             ))),
```

Comment:
It seems this is not really related to `metasrv`.

`UserStageInfo` is not used in `metasrv`, by leaving the `type Error` a `ErrorCode`, it may be more compatible with the style in crate `query`.

Reply:
So is Compression and Format.

IMHO it does not need to modify the error types for these type.